### PR TITLE
Fixes #14147 - Allow repo names in creating content view filters

### PR DESCRIPTION
**Pull request by akofink**
Continuation of #362

### CODE REVIEW

**Comment by akofink** [comment]
@ehelms Any more tests that might be relevant to these changes?

File: lib/hammer_cli_katello/id_resolver.rb [hunk 1]
```diff
@@ -38,6 +38,14 @@ def for(resource)
   end
 
   class IdResolver < HammerCLIForeman::IdResolver
+    require 'hammer_cli_katello/search_options_creators'
```
Comment (ehelms) [inline lib/hammer_cli_katello/id_resolver.rb]: This seem strange to need to do here in general and to do it at least at the top of the file if required.

File: lib/hammer_cli_katello/id_resolver.rb [hunk 2]
```diff
@@ -38,6 +38,14 @@ def for(resource)
...
+    include HammerCLIKatello::SearchOptionsCreators
+
+    # alias_method_chain :create_search_options, :katello_api
+    alias_method :create_search_options_without_katello_api,
+      :create_search_options
+    alias_method :create_search_options,
+      :create_search_options_with_katello_api
```
Comment (ehelms) [inline lib/hammer_cli_katello/id_resolver.rb]: Did these two break the 120 character mark?

**Comment by ehelms** [comment]
Tests look great!

**Comment by jlsherrill** [comment]
@akofink this doesn't appear to fix the original issue:

```
hammer content-view  filter create --content-view=test  --organization="Default Organization"  --repositories=test --type rpm  --name=foo3 
```

seemed to select more than just the 'test' repo when creating the filter

**Comment by akofink** [comment]
Merge with https://github.com/Katello/katello/pull/5950

**Comment by akofink** [comment]
@ehelms The original issue should be fixed now.

**Comment by komidore64** [comment]
@akofink did rubocop yell about a class being too large? Is that why you split out ``search_options_creators.rb``?

Does it make more sense to have one ``search_options_creators.rb`` file for all of them or a ``<thing>_options_creators.rb`` for each creator? That way we could keep the modules small? Or is that not worth it?

**Comment by akofink** [comment]
@tstrachota Yes, I admit I separated it for rubocop.

As for splitting into smaller modules, there is common functionality for a few of the creators, such as [L39](https://github.com/akofink/hammer-cli-katello/blob/14147/lib/hammer_cli_katello/search_options_creators.rb#L39),[L43](https://github.com/akofink/hammer-cli-katello/blob/14147/lib/hammer_cli_katello/search_options_creators.rb#L43),[L47](https://github.com/akofink/hammer-cli-katello/blob/14147/lib/hammer_cli_katello/search_options_creators.rb#L47),[L51](https://github.com/akofink/hammer-cli-katello/blob/14147/lib/hammer_cli_katello/search_options_creators.rb#L51). I think this shared logic would complicate the effort of splitting them into smaller modules, though I'm open to discussion of course.

**Comment by komidore64** [comment]
One file is good for now. I'm was just curious if that was why.

Also: it looks like you'll need to rebase with upstream.

ACK.

**Comment by akofink** [comment]
@komidore64 rebased.

**Comment by jlsherrill** [comment]
Seems to resolve the original issue, ACK from a functionality perspective

**Comment by jlsherrill** [comment]
Merging  based on my ACK and @komidore64 ACK  Thanks @akofink

**Pull request merged**